src/phlorest/nexuslib.py

```diff
@@ -86,3 +88,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 nex.trees.append(NexusTree.from_newick(
                     tree.newick_string, name=tree.name or 'tree{}'.format(i), rooted=tree.rooted))
             nex.write_to_file(self.path)
+            if self.zipped:
+                with zipfile.ZipFile(
+                    self.path.parent / (self.path.name + '.zip'),
+                    'w',
+                    compression=zipfile.ZIP_DEFLATED
+                ) as zf:
```

Comment:
is it worth setting `compresslevel` to 9? (I think zipfile defaults to 6)

Reply:
I don't have real evidence here, but I think compression/decompression time is relevant for us, too, so a level (6) that's chosen as default time-vs-size compromise sounds ok for me - unless we do real profiling to base a decision on.

Reply:
hmm. a quick benchmark seems to show no difference (on a large file -- gray_et_al's 2009 posterior.trees:

```
read 6: 0.0886 (0.0018)
read 9: 0.0874 (0.0009)
write 6: 0.0881 (0.0014)
write 9: 0.0894 (0.0013)
```

...but not much difference in filesize either:

```shell
-rw-r--r--   1 simon  staff   4074565 Nov 28 10:17 posterior.trees.6.zip
-rw-r--r--   1 simon  staff   4018583 Nov 28 10:17 posterior.trees.9.zip
```

Reply:
benchmark code:

```python
import timeit
import tempfile
from statistics import median, stdev
from zipfile import ZipFile
from functools import partial

def read(filename):
    with ZipFile(filename) as z:
        with z.open('posterior.trees') as h:
            h.read()

def write(compresslevel):
    with tempfile.TemporaryDirectory() as tmp:
        with ZipFile(tmp / 'out.zip', 'w', compresslevel=compresslevel) as z:
            z.write('posterior.trees')


rf6 = partial(read, 'posterior.trees.6.zip')
rf9 = partial(read, 'posterior.trees.9.zip')

wf6 = partial(write, 6)
wf9 = partial(write, 9)


r6 = timeit.Timer(rf6).repeat(repeat=10, number=3)
r9 = timeit.Timer(rf9).repeat(repeat=10, number=3)


print("read 6: %0.4f (%0.4f)" % (median(r6), stdev(r6)))
print("read 9: %0.4f (%0.4f)" % (median(r9), stdev(r9)))

w6 = timeit.Timer(rf6).repeat(repeat=10, number=3)
w9 = timeit.Timer(rf9).repeat(repeat=10, number=3)

print("write 6: %0.4f (%0.4f)" % (median(w6), stdev(w6)))
print("write 9: %0.4f (%0.4f)" % (median(w9), stdev(w9)))
```

Reply:
So, I'd leave the default.

Reply:
yep, no benefit for the cost.